near-token-factory/src/lib.rs

```diff
@@ -42,39 +54,60 @@ pub struct Contract {
 }
 
 // TODO: Add pausable
-// TODO: Add access control
 #[near_bindgen]
 impl Contract {
     /// Initializes the contract. The locker account id MUST be the NEAR
     /// representative of the Aurora address of the locker contract created
     /// using the Cross Contract Call interface.
     #[init]
-    pub fn new(aurora: AccountId, locker: aurora_sdk::Address) -> Self {
+    pub fn new(aurora: AccountId, locker: aurora_sdk::Address, admin: Option<AccountId>) -> Self {
         require!(
             env::current_account_id().as_str().len() + 1 + 40 <= 63,
             ERR_INVALID_ACCOUNT
         );
 
-        Self {
+        // If not specified, the admin is the deployer of this contract.
+        let admin = admin.unwrap_or_else(env::predecessor_account_id);
+
+        let mut contract = Self {
             aurora,
+            token_admin: admin.clone(),
             token_binary: LazyOption::new(StorageKey::TokenBinary, None),
             token_binary_version: 0,
             tokens: UnorderedMap::new(StorageKey::TokenMap),
             locker,
-        }
+            __acl: Default::default(),
+        };
+
+        // Make the factory acl super-admin and grant roles to it.
```

Comment:
```suggestion
        // Initialize Acl permissions.
```

Since there's no factory variable in scope.

Reply:
Done in ccd84c46af2278ef9b81fb21dcfa824d28ea41c6